gst/edge/edge_sink.c

```diff
@@ -295,6 +322,17 @@ gst_edgesink_start (GstBaseSink * basesink)
     return FALSE;
   }
 
+  if (self->wait_connection) {
+    guint64 counter = 0, timeout = self->connection_timeout;
+    if (0 == timeout)
+      timeout = G_MAXUINT64;
+
+    while ((timeout / 10) > counter++ &&
+        NNS_EDGE_ERROR_NONE != nns_edge_is_connected (self->edge_h)) {
+      g_usleep (10000);
```

Comment:
Let's modify the waiting logic. It's too coarse.

```
/* in if(self->wait_connection) */

guint64 remaining = timeout;
while (remaining >= 10 && ....) {
  g_usleep(10000);
  remaining -= 10;
}
if (remaining > 0 && not connected) {
  g_usleep (remaining * 1000U);
  retry once..
}

if (not connected)
   error_once.... some message.

```

Reply:
Thanks, I added to check `self->wait_connection` in the loop.